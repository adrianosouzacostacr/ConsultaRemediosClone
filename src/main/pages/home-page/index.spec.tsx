import 'react-native';
import React from 'react';
import HomePage from '.';

import {render, cleanup} from '@testing-library/react-native';

describe('Home page', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const {toJSON} = render(<HomePage />);
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display correctly infos', () => {
    const {getByA11yLabel} = render(<HomePage />);
    const messageLabel = getByA11yLabel('message');
    const brandLabel = getByA11yLabel('brand');

    expect(messageLabel.children[0]).toBe('Agora sim vai dar certo \\o/');
    expect(brandLabel.children[0]).toBe('Consulta Rem√©dios');
  });
});
