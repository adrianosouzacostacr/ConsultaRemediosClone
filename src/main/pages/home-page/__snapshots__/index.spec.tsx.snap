// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "overflow": "visible",
    }
  }
>
  <Text
    accessibilityLabel="brand"
    testID=""
  >
    Consulta Rem√©dios
  </Text>
  <Text
    accessibilityLabel="message"
    testID=""
  >
    Agora sim vai dar certo \\o/
  </Text>
</RCTSafeAreaView>
`;
